Core/Core/Api/GraphQL/GraphQLHttpClientExtensions.cs

```diff
@@ -0,0 +1,47 @@
+using GraphQL;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Threading;
+using GraphQL.Client.Http;
+
+namespace Speckle.Core.Api.GraphQL;
+
+public static class GraphQLHttpClientExtensions
+{
+  /// <summary>
+  /// Gets the version of the current server. Useful for guarding against unsupported api calls on newer or older servers.
+  /// </summary>
+  /// <param name="cancellationToken">[Optional] defaults to an empty cancellation token</param>
+  /// <returns><see cref="Version"/> object excluding any strings (eg "2.7.2-alpha.6995" becomes "2.7.2.6995")</returns>
+  /// <exception cref="SpeckleGraphQLException"></exception>
+  public static async Task<System.Version> GetServerVersion(
+    this GraphQLHttpClient client,
+    CancellationToken cancellationToken = default
+  )
+  {
+    var request = new GraphQLRequest
+    {
+      Query =
+        @"query Server {
+                    serverInfo {
+                        version
+                      }
+                  }"
+    };
+
+    var response = await client.SendQueryAsync<ServerInfoResponse>(request, cancellationToken).ConfigureAwait(false);
+
+    if (response.Errors != null)
+    {
+      throw new SpeckleGraphQLException<ServerInfoResponse>(
+        $"Query {nameof(GetServerVersion)} failed",
+        request,
+        response
+      );
+    }
+
+    return response.Data.serverInfo.version.Contains("dev")
+      ? new System.Version(999, 999, 999)
+      : new System.Version(Regex.Replace(response.Data.serverInfo.version, "[-a-zA-Z]+", ""));
```

Comment:
I think you need to run me through this 🤔 If we're checking if it contains `dev`, why are we then removing all A-Z characters with regex.

Do we know if `dev` is the only possible edge-case?

Reply:
This logic was taken from Client.ServerOperations.cs

Unfortunately, I could just use that implementation because it only applies to a Speckle.Core.Api.Client which is coupled to our Account class. Where this method is being used currently is a static context that is ignorant of any accounts so we can't use the existing logic.

Reply:
So you're saying we now have this weird logic in 2 places? 😅

I think we should check with someone from the server team what are the edge-cases expected in the serverInfo.version return value.

Reply:
Can we have a quick chat on this today? It would be nice if we could avoid duplicating this logic

Reply:
From our discussion with @didimitrie and @gjedlicska :
 - The other server version check can be killed, was only for 2.6.0 backwards compact
 - The `.Contains("Dev")` can be replaced by a `== "Dev"`

There's a couple other topics I want chat about next week in Prague r.e. our `Client` class. But for now, just the above ^ affects this PR.